Source/RimworldMod/Comp/ShipHeatMapComp.cs

```diff
@@ -268,7 +268,51 @@ public void SpawnEnemyShip(PassingShip passingShip)
             bool isDerelict = false;
             EnemyShipDef shipDef;
 
-            if (passingShip is AttackableShip)
+            // Set ship faction
+
+            // Find an enemy space navy
+            Faction enemyShipFaction = Faction.OfAncientsHostile;
+
+            IEnumerable<SpaceNavyDef> spaceNavyDefs = DefDatabase<SpaceNavyDef>.AllDefs;
+
+            SpaceNavyDef enemySpaceNavyDef = null;
+            EnemyShipDef enemySpaceNavyShipDef = null;
+
+            System.Random s_Random = new System.Random();
+
+            if (s_Random.Next(0, 100) <= ShipInteriorMod2.navyEncounterChance)
+            {
+                // Look through space navies
+                foreach (SpaceNavyDef spaceNavyDef in spaceNavyDefs)
+                {
+                    Faction spaceNavyFaction = Find.FactionManager.AllFactions.Where(faction => faction.def == spaceNavyDef.factionDef).RandomElement();
```

Comment:
With this you'll be getting the same space navy over and over again, a solution would be to get all the hostile space navies into a List, and choose one randomly.